### ISSUE TITLE
SSL Error, but only in a Fibre, and only on ARM

### ISSUE THREAD

**Issue by HCLarsen** [crystal-lang/crystal#10040]
I'm getting a very unusual error that only happens within very specific circumstances. The error is this:

```
Unhandled exception in spawn: SSL_connect: error:1417B07B:SSL routines:tls_process_cert_verify:bad signature (OpenSSL::SSL::Error)
  from ../../snap/crystal/539/share/crystal/src/openssl/ssl/socket.cr:34:11 in 'initialize'
  from ../../snap/crystal/539/share/crystal/src/openssl/ssl/socket.cr:3:5 in 'new:context:sync_close:hostname'
  from ../../snap/crystal/539/share/crystal/src/http/client.cr:784:5 in 'io'
  from ../../snap/crystal/539/share/crystal/src/http/client.cr:664:5 in 'send_request'
  from ../../snap/crystal/539/share/crystal/src/http/client.cr:599:5 in 'exec_internal_single'
  from ../../snap/crystal/539/share/crystal/src/http/client.cr:586:5 in 'exec_internal'
  from ../../snap/crystal/539/share/crystal/src/http/client.cr:581:7 in 'exec'
  from ../../snap/crystal/539/share/crystal/src/http/client.cr:706:5 in 'exec'
  from ../../snap/crystal/539/share/crystal/src/http/client.cr:738:7 in 'exec'
  from ../../snap/crystal/539/share/crystal/src/http/client.cr:406:3 in 'get'
  from ../clarsen/workspace/test/ssltest.cr:8:3 in '->'
  from ../../snap/crystal/539/share/crystal/src/primitives.cr:255:3 in 'run'
  from ../../snap/crystal/539/share/crystal/src/fiber.cr:92:34 in '->'
```

The error, as I said, only occurs under very specific circumstances. This code is what gave this error:

```
require "http"

spawn do
  puts "Begun"
  HTTP::Client.get("https://restcountries.eu/rest/v2/alpha/can")
  puts "Finished"
end

sleep
```

It's actually very specific to that code. If I remove the fibre, and run the code in the main thread, it throws no error. If I put `puts` in front of the HTTP client, it doesn't throw the error, and if I remove the `puts "Finished` line, it also doesn't throw the error.

Also, this error is only produced when I cross compile it and run it on Ubuntu Server on a Raspberry Pi 2B. It doesn't give an error on my Mac, nor on Ubuntu running in a VM. The command I used to cross-compile is this:

```
crystal build ssltest.cr --cross-compile --target "arm7l-unknown-linux-gnueabihf"
```

I know ARM isn't fully supported, but when I discussed this in the forum, it was suggested that I bring this up here to help bring us closer to full ARM support. It's also important to note that the SSL error changes sometimes. At one point it was `error:0D0C5006:asn1`, but whenever I run it today, whether using the nightly build, or the official build of Crystal, I'm getting `error:1417B07B:SSL`, as noted above.

**Comment by jhass** [comment]
This is most likely some stack corruption. Given it happens inside a C library, most likely something wrong with the ABI, perhaps #9533

**Comment by straight-shoota** [comment]
For reference, this was originally posted to the forum: https://forum.crystal-lang.org/t/ssl-error-but-only-in-a-fibre/2718

**Comment by HCLarsen** [comment]
Thanks @straight-shoota, I should have mentioned that.

**Comment by beta-ziliani** [comment]
I'm closing this issue since I can't reproduce it in 1.3.0-dev. Probably related to the already fixed #9533, as mentioned by @jhass .
